pkg/workflows/wasm/host/module.go

```diff
@@ -451,12 +460,19 @@ func createFetchFn(
 			return ErrnoFault
 		}
 
-		storedRequest, innerErr := store.get(req.Id)
+		storedRequest, innerErr := requestStore.get(req.Id)
 		if innerErr != nil {
 			logger.Errorf("%s: %s", errFetchSfx, innerErr)
 			return ErrnoFault
 		}
 
+		// limit the number of fetch calls we can make per request
+		if storedRequest.counter >= modCfg.MaxFetchRequests {
+			logger.Errorf("%s: max number of fetch request %d exceeded", errFetchSfx, modCfg.MaxFetchRequests)
+			return ErrnoFault
+		}
+		defer func() { storedRequest.counter++ }()
```

Comment:
You don't really need to defer this; just increment it after you've checked it :)